### DISCUSSION TITLE
Missing artistID values on tracks for "Various Artists" albums

### DISCUSSION THREAD

**Discussion by khers** [navidrome/navidrome#3147] [Q&A] [answered]
I work on the Music Assistant project and a use mentioned a problem when the try and access their Navidrome library. After investigating the cause I discovered that when Navidrome returns a Song object which is on an album with "Various Artists" that the artist name field will have a value but the artistId field is left null. Further, when I search for one of these artist names, I get no artist results, only albums and songs. Is this expected behavior?

Example:
```
{
  "id": "0017034d06f59511dee9eb0677917165",
  "coverId": "mf-0017034d06f59511dee9eb0677917165_64d42e53",
  "starred": null,
  "parent": "9074c7e80584c31fb2bef427e78d58c7",
  "title": "Into the Mire \u2013 Stillwind",
  "album": "FINAL FANTASY XVI Original Soundtrack Ultimate Edition",
  "albumId": "9074c7e80584c31fb2bef427e78d58c7",
  "artist": "Masayoshi Soken",
  "displayArtist": null,
  "displayAlbumArtist": null,
  "artistId": null,
  "isDir": false,
  "created": "2023-08-09T20:51:39.4850143-04:00",
  "duration": 288,
  "bitRate": 216,
  "size": 7908172,
  "suffix": "m4a",
  "contentType": "audio/mp4",
  "isVideo": false,
  "path": "Various Artists/FINAL FANTASY XVI Original Soundtrack Ultimate Edition/08 - Into the Mire \u2013 Stillwind.m4a",
  "track": 8,
  "type": "music",
  "year": 2023
}
```
And
```
>>> art = conn.search3('Masayoshi Soken')
>>> art
{'artists': [], 'albums': [<libopensonic...
```

**Comment by metalheim** [comment, marked as answer]
> Further, when I search for one of these artist names, I get no artist results, only albums and songs. Is this expected behavior?

Yes indeed. Navidrome currently only indexes artists that have at least 1 album of their own (where they are the albumartist) as a real, dedicated artist.
Since they arent indexed as artists, they dont have an artistid of their own.
See issue #211 for more details

deluan is currently working on the big refactoring of the scanner and database layout, so this behaviour might change after the BFR (Big refactor) is finished. I think one goal of the BFR is to import all artists, even if they only have a single song participation, but no album of their own.

**Regarding "Various Artists":**
There are different philosophies for this, but I think navidrome handles that gracefully by not giving out any name or id and letting the client figure out how to display various artists.
One client might want to show a translated version of "Various Artists" to the users, another might want to have "VA". So by not returning it, the client is free to implement what they want without needed to hardcode anything in.
You could argue that Navidrome should at least return the field with a `null` value or something to make this more clear

**Reply by khers** [reply]
Thanks for the response, I will try to work around this in the subsonic provider for MA.
